lib/select.js

```diff
@@ -53,7 +53,7 @@ global.select = factory();
   }
 
   function activeElement(element) {
-    return element.shadowRoot ? activeElement(element.shadowRoot.activeElement) : element;
+    return element.shadowRoot && element.shadowRoot.activeElement ? activeElement(element.shadowRoot.activeElement) : element;
```

Comment:
Dus vroeger bleef hij recursief loopen tot er geen shadowRoot maar was. Nu tot er geen activeElement meer op de shadowRoot zit. Maar geeft deze methode dan het 'active element' terug of het onderliggende?

Reply:
@Pieter-Beckers  heeft deze beantwoord in een aparte comment, zie:
https://github.com/milieuinfo/webcomponent-vl-ui-select/pull/166#discussion_r675348180

Met "Nu dus het bovenliggende activeElement ipv een exception".

"The activeElement read-only property of the Document interface returns the Element within the DOM that currently has focus."
"return value: The Element which currently has focus, <body> or null if there is no focused element."
src: https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement

Als goed lees vermijden we hier die null pointer. Nu of dit al dan niet gewenst gedrag is geweest dat kan ik niet zeggen. Er staat namelijk geen documentatie op deze methode. shadowRoot.activeElement geeft namelijk ook 'null' terug maar daarvan staat dit ook zo neergeschreven als verwacht gedrag. Aangezien er hier dezelfde benaming gebruikt wordt kan je dat hier miss ook verwachten en moet de aanroeper dit counteren (niet de methode zelf).

Nu kwa algoritme blijven we dus zoeken tot er geen onderliggen active element meer beschikbaar is en indien zo geven we dit terug. Ik denk dus dat wat Pieter zegt niet volledig klopt.

Reply:
@LucasSledsens we zullen deze eens samen bespreken

Reply:
Ben even gaan zoeken waar deze code gebruikt wordt en in select.js komt het 9 maal voor.
Waarbij meestal in dit soort vergelijkingen:
`this.input === activeElement(document.activeElement),`